consensus/core/src/authority_service.rs

```diff
@@ -473,31 +514,30 @@ impl<C: CoreThreadDispatcher> NetworkService for AuthorityService<C> {
                     continue;
                 }
 
-                let missing_blocks = dag_state.get_cached_blocks_in_range(
-                    authority,
-                    highest_accepted_round + 1,
-                    lowest_missing_round,
-                    self.context
-                        .parameters
-                        .max_blocks_per_sync
-                        .saturating_sub(blocks.len()),
-                );
-                blocks.extend(missing_blocks);
-                if blocks.len() >= self.context.parameters.max_blocks_per_sync {
-                    blocks.truncate(self.context.parameters.max_blocks_per_sync);
+                let missing_blocks = dag_state
+                    .get_cached_blocks_in_range(
```

Comment:
why not adding the faulty ones too?

Reply:
Just because we don't store cached faulty blocks by authority and round as we do with verified blocks, so it is not possible at present to retrieve the missing ones in this way.